packages/components/src/button/style.scss

```diff
@@ -73,39 +73,43 @@
 			-1px 0 1px color(theme(button) shade(30%));
 
 		&:hover,
-		&:focus:not(:disabled):not([aria-disabled="true"]) {
+		&:focus:enabled {
 			background: color(theme(button) shade(5%));
 			border-color: color(theme(button) shade(50%));
-			box-shadow: inset 0 -1px 0 color(theme(button) shade(50%));
 			color: $white;
 		}
 
-		&:focus:not(:disabled):not([aria-disabled="true"]) {
+		&:hover {
+			box-shadow: inset 0 -1px 0 color(theme(button) shade(50%));
+		}
+
+		&:focus:enabled {
 			box-shadow:
 				inset 0 -1px 0 color(theme(button) shade(50%)),
 				0 0 0 2px $blue-medium-200;
 		}
 
-		&:active:not(:disabled):not([aria-disabled="true"]) {
+		&:active:enabled {
 			background: color(theme(button) shade(20%));
 			border-color: color(theme(button) shade(50%));
 			box-shadow: inset 0 1px 0 color(theme(button) shade(50%));
 			vertical-align: top;
 		}
 
 		&:disabled,
-		&[disabled] {
-			color: color(theme(button) tint(30%)) !important;
-			background: color(theme(button) shade(30%)) !important;
-			border-color: color(theme(button) shade(20%)) !important;
-			box-shadow: none !important;
-			text-shadow: 0 -1px 0 rgba(0, 0, 0, 0.1) !important;
+		&[aria-disabled="true"] {
+			color: color(theme(button) tint(30%));
+			background: color(theme(button) shade(30%));
+			border-color: color(theme(button) shade(20%));
+			box-shadow: none;
+			text-shadow: 0 -1px 0 rgba(0, 0, 0, 0.1);
 		}
 
 		&.is-busy,
-		&.is-primary.is-busy[disabled] {
-			color: $white !important;
-			background-size: 100px 100% !important;
+		&.is-busy:disabled,
+		&.is-busy[aria-disabled="true"] {
```

Comment:
Note: these two are losing the `&.is-primary` selector because we are already nested inside `&.is-primary {}` (line 63).